<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SokarSegmentationPipelineEditor</class>
 <widget class="QWidget" name="SokarSegmentationPipelineEditor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>900</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>400</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Interpolacja</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_4">
      <item row="0" column="0">
       <widget class="QLabel" name="label_10">
        <property name="text">
         <string>Bok woksela [mm]</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Wielkość Y</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Wielkość X</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QtUi::SpinBox" name="interpolationSizeX"/>
      </item>
      <item row="3" column="1">
       <widget class="QtUi::SpinBox" name="interpolationSizeY"/>
      </item>
      <item row="0" column="1">
       <widget class="QtUi::DoubleSpinBox" name="interpolationWokselSize">
        <property name="value">
         <double>3.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="1" column="0" colspan="2">
       <widget class="QtUi::ComboBox" name="interpolationCombo">
        <property name="currentIndex">
         <number>0</number>
        </property>
        <item>
         <property name="text">
          <string>Najbliższa</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Liniowa</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Wielomianowa</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Wielomioanowa (C++11)</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Sześciennna</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Calum Rom</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QtUi::SpinBox" name="interpolationSizeZ"/>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Wielkość Z</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QCheckBox" name="cacheInterpolation">
        <property name="text">
         <string notr="true"/>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="label_8">
        <property name="text">
         <string>Cache</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Segmentacja</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="3" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Wartość dolna</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QtUi::DoubleSpinBox" name="segmentationTresholdUp">
        <property name="minimum">
         <double>-65536.000000000000000</double>
        </property>
        <property name="maximum">
         <double>65536.000000000000000</double>
        </property>
        <property name="value">
         <double>100.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="0" column="0" colspan="2">
       <widget class="QtUi::ComboBox" name="segmentationAlgorithm">
        <property name="currentIndex">
         <number>0</number>
        </property>
        <item>
         <property name="text">
          <string>Maszerujące sześciany</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Maszerujące czworościany</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Pamięć podręczna</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="label_12">
        <property name="text">
         <string>Rozrost obszaru</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QtUi::DoubleSpinBox" name="segmentationTresholdDown">
        <property name="minimum">
         <double>-65536.000000000000000</double>
        </property>
        <property name="maximum">
         <double>65536.000000000000000</double>
        </property>
        <property name="value">
         <double>20000.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="7" column="0">
       <widget class="QLabel" name="label_13">
        <property name="text">
         <string>Śiatka trójkątów</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QCheckBox" name="regionGrowthCheck">
        <property name="text">
         <string notr="true"/>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Wartość górna</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QtUi::ComboBox" name="useCache">
        <item>
         <property name="text">
          <string>Tak</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Nie</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="7" column="1">
       <widget class="QtUi::ComboBox" name="meshTypeCombo">
        <item>
         <property name="text">
          <string>Lista trójkątów</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Indeksowana</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Indkesowana z podziałem obszarów</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="6" column="0" colspan="2">
       <widget class="QtUi::Vec3Input" name="regionGrowthVec" native="true"/>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_4">
     <property name="title">
      <string>Uproszczenia siatki</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_5">
      <item>
       <widget class="QtUi::ComboBox" name="simplificationAlgorithm">
        <item>
         <property name="text">
          <string>Brak</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Klasteryzacja wierzchołków</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Usuwanie krawędzi</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Bounding convex decomposition</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="vertexClustering" native="true">
        <layout class="QGridLayout" name="vertexClusteringLayout">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item row="0" column="0" colspan="2">
          <widget class="QtUi::Vec3Input" name="vertexClusteringSize" native="true"/>
         </item>
         <item row="1" column="0" colspan="2">
          <widget class="QtUi::Vec3Input" name="vertexClusteringOffset" native="true"/>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="edgeCollapse" native="true">
        <layout class="QGridLayout">
         <item row="1" column="1">
          <widget class="QtUi::ComboBox" name="edgeCollapseDirection">
           <item>
            <property name="text">
             <string>Brak</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Do wewnątrz</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Na zewnątrz</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_14">
           <property name="text">
            <string>Błąd</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_11">
           <property name="text">
            <string>Kierunek</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QtUi::DoubleSpinBox" name="edgeCollapseError">
           <property name="value">
            <double>10.000000000000000</double>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_15">
           <property name="text">
            <string>Procent redukcji</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QtUi::DoubleSpinBox" name="edgeCollapseVertReduction">
           <property name="value">
            <double>50.000000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <layout class="QGridLayout" name="gridLayout_3"/>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_3">
     <property name="title">
      <string>Wygląd</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="1" column="0">
       <widget class="QLabel" name="label_9">
        <property name="text">
         <string>Osadz w pustej przestrzeni</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout" stretch="1,0">
        <item>
         <widget class="QPushButton" name="pickColorButton">
          <property name="text">
           <string>PushButton</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="randomColorButton">
          <property name="text">
           <string notr="true"/>
          </property>
          <property name="icon">
           <iconset theme="reload">
            <normaloff>.</normaloff>.</iconset>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="1" column="1">
       <widget class="QCheckBox" name="emptyEnv">
        <property name="text">
         <string notr="true"/>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label_7">
        <property name="text">
         <string>Kolor</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QtUi::Vec3Input</class>
   <extends>QWidget</extends>
   <header>QtUi/Vec3Input.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QtUi::DoubleSpinBox</class>
   <extends>QDoubleSpinBox</extends>
   <header>QtUi/DoubleSpinBox.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QtUi::SpinBox</class>
   <extends>QSpinBox</extends>
   <header>QtUi/SpinBox.hpp</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QtUi::ComboBox</class>
   <extends>QComboBox</extends>
   <header>QtUi/ComboBox.hpp</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>interpolationWokselSize</tabstop>
  <tabstop>interpolationCombo</tabstop>
  <tabstop>interpolationSizeX</tabstop>
  <tabstop>interpolationSizeY</tabstop>
  <tabstop>interpolationSizeZ</tabstop>
  <tabstop>cacheInterpolation</tabstop>
  <tabstop>segmentationAlgorithm</tabstop>
  <tabstop>segmentationTresholdUp</tabstop>
  <tabstop>segmentationTresholdDown</tabstop>
  <tabstop>useCache</tabstop>
  <tabstop>regionGrowthCheck</tabstop>
  <tabstop>meshTypeCombo</tabstop>
  <tabstop>simplificationAlgorithm</tabstop>
  <tabstop>pickColorButton</tabstop>
  <tabstop>randomColorButton</tabstop>
  <tabstop>emptyEnv</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
